src/main/java/com/alipay/oceanbase/hbase/OHTable.java

```diff
@@ -1210,6 +1173,47 @@ private ObTableBatchOperation buildObTableBatchOperation(List<KeyValue> keyValue
         return batch;
     }
 
+    private com.alipay.oceanbase.rpc.mutation.Mutation buildMutation(KeyValue kv, boolean putToAppend) {
+        KeyValue.Type kvType = KeyValue.Type.codeToType(kv.getType());
+        switch (kvType) {
+            case Put:
+                ObTableOperationType operationType;
+                if (putToAppend) {
+                    operationType = APPEND;
+                } else {
+                    operationType = INSERT_OR_UPDATE;
+                }
+                return com.alipay.oceanbase.rpc.mutation.Mutation.getInstance(operationType, ROW_KEY_COLUMNS,
+                        new Object[]{kv.getRow(), kv.getQualifier(), kv.getTimestamp()}, V_COLUMNS,
+                        new Object[]{kv.getValue()});
+            case Delete:
+                return com.alipay.oceanbase.rpc.mutation.Mutation.getInstance(DEL, ROW_KEY_COLUMNS,
+                        new Object[]{kv.getRow(), kv.getQualifier(), kv.getTimestamp()}, null, null);
+            case DeleteColumn:
+                return com.alipay.oceanbase.rpc.mutation.Mutation.getInstance(DEL, ROW_KEY_COLUMNS,
+                        new Object[]{kv.getRow(), kv.getQualifier(), -kv.getTimestamp()}, null, null);
+            case DeleteFamily:
+                return com.alipay.oceanbase.rpc.mutation.Mutation.getInstance(DEL, ROW_KEY_COLUMNS, new Object[]{kv.getRow(), null, -kv.getTimestamp()},
+                        null, null);
+            default:
+                throw new IllegalArgumentException("illegal mutation type " + kvType);
+        }
+    }
+
+    private BatchOperation buildBatchOperation(String tableName, List<KeyValue> keyValueList,
+                                               boolean putToAppend,
+                                               List<byte[]> qualifiers) {
+        BatchOperation batch = obTableClient.batchOperation(tableName);
+        for (KeyValue kv : keyValueList) {
+            if (qualifiers != null) {
+                qualifiers.add(kv.getQualifier());
+            }
+            batch.addOperation(buildMutation(kv, putToAppend));
+        }
+        batch.setEntityType(ObTableEntityType.HKV);
```

Comment:
What is the atomicity of the HBase request? Could we set this attribute?

Reply:
As before the modification, atomicity uses the default attributes in table_client. The old format is false by default, and the new format is true by default.I don't think it's necessary to modify this attribute.